nixos/modules/services/web-apps/immich.md

```diff
@@ -0,0 +1,265 @@
+[Configuring](#module-immich-configuring).
+
+After building and switching to your new configuration, you should
+stop the newly started `immich-server.service` and
+`immich-machine-learning.service` units.
+
+```console
+# systemctl stop immich-server.service immich-machine-learning.service
+```
+
+### Create database dump {#module-immich-migrating-from-docker-compose-dump-database}
+
+You might have done this already during your
+[backup step](#module-immich-migrating-from-docker-compose-backup).
+You can choose to reuse that database dump by making a copy, or just
+create another dump like this:
+
+```console
+# docker exec immich_postgres pg_dump --username immich immich -f /tmp/immich.sql
+# docker cp immich_postgres:/tmp/immich.sql ./immich.sql
+```
+
+::: {.note}
+While `pg_dump` offers a few output formats, this guide relies on it
+being `plain` (default), as we will have to modify it later.
+:::
+
+### Fix up database dump {#module-immich-migrating-from-docker-compose-fixup-dump}
+
+The database dump generated by `pg_dump`, can most likely not be
+imported as is. There are two changes we need to make, to allow it to
+be imported correctly.
+
+First, we need to fix the `search_path` variable of the dump, as a bug
+in PostgreSQL's `earthdistance` extension seems to cause issues when
+importing database dumps that contain its units.
+(See [this StackOverflow comment](https://stackoverflow.com/questions/49702964/postgres-import-to-rds-could-not-execute-query-error-type-earth-does-not-ex#comment86441215_49702964))
+To work around this issue, we can adjust the `search_path` variable
+in the database dump. This variable is set within the first 20
+lines of your database dump, so you can just open up your database
+dump in your preferred text editor and apply the following changes:
+
+```diff
+--- a/immich.sql
++++ b/immich.sql
+@@ -10,7 +10,7 @@ SET lock_timeout = 0;
+ SET idle_in_transaction_session_timeout = 0;
+ SET client_encoding = 'UTF8';
+ SET standard_conforming_strings = on;
+-SELECT pg_catalog.set_config('search_path', '', false);
++SELECT pg_catalog.set_config('search_path', 'public', false);
+ SET check_function_bodies = false;
+ SET xmloption = content;
+ SET client_min_messages = warning;
+```
+
+The next issue we will need to work around is that Immich' Docker
+deployment caused all your image files to be referenced relative to
+its working directory (`/usr/src/app` inside the container). This
+means that all of your photos, videos and thumbnails are going to
+appear with the path prefix `upload/` in the database dump, as
+immich stores its data in `./upload` relative to its working directory
+by default. See
+[Immich "Files Custom Locations" documentation](https://immich.app/docs/guides/custom-locations/)
+for more information.
+
+In order to fix these file paths in our database dump, we can use the
+following `sed` command:
+
+```console
+# sed "s#upload/(.*)/#$TARGET_LOCATION/\1/#g" ./immich.sql > ./immich-new-path.sql
```

Comment:
I may not understand this, but is this the total path? Like for instance:
`upload/image.png`
should be sed'd to:
`/var/lib/immich/photos/image.png`?
so TARGET_LOCATION should be:
`/var/lib/immich/photos`

correct me if im wrong. Or is this relative to `mediaLocation`? I know this is explained but maybe provide an example :p

Reply:
It seems that Immich stores paths based on the value of `IMMICH_MEDIA_LOCATION`. For Docker deployments that value was `./upload`, and for NixOS it's going to be `$TARGET_LOCATION`

So we want to replace the relative `upload/` path with an absolute `$TARGET_LOCATION`.

Reply:
I have expanded on this a little bit in the documentation. I also added another assumption up top that `TARGET_PATH` is absolute